src/api/routes/ws/socket-io.ts

```diff
@@ -64,6 +94,21 @@ export function createSocketIORouter(db: DataStore, server: http.Server) {
     });
   });
 
+  io.use((socket, next) => {
+    const subscriptions = socket.handshake.query['subscriptions'];
+    if (subscriptions) {
+      const topics = [...[subscriptions]].flat().flatMap(r => r.split(','));
+      if (!areSubscriptionsValid(topics as Topic[])) {
+        const error = new Error('Invalid topic');
+        next(error);
+      } else {
```

Comment:
In order to be able to tell the user which topic was invalid, you should iterate and validate the topics one by one so you can do:
```ts
next(new Error(`Invalid subscription: ${topic}`);
```